llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -220,6 +220,13 @@ def idempotent_prop : GICombineRule<
    (match (idempotent_prop_frags $dst, $src)),
    (apply (GIReplaceReg $dst, $src))>;
 
+// Convert freeze(Op(Op0, NonPoisonOps...)) to Op(freeze(Op0), NonPoisonOps...)
+// when Op0 is not guaranteed non-poison
+def push_freeze_to_prevent_poison_propagation : GICombineRule<
```

Comment:
maybe change to more closely match InstCombine's pushFreezeToPreventPoisonFromPropagating? i.e. push_freeze_to_prevent_poison_from_propagating